src/hotspot/share/opto/macro.cpp

```diff
@@ -2563,10 +2563,7 @@ void PhaseMacroExpand::eliminate_macro_nodes() {
   bool progress = true;
   while (progress) {
     progress = false;
-    for (int i = C->macro_count(); i > 0; i--) {
-      if (i > C->macro_count()) {
-        i = C->macro_count(); // more than 1 element can be eliminated at once
-      }
+    for (int i = C->macro_count(); i > 0; i = MIN2(i-1, C->macro_count())) { // more than 1 element can be eliminated at once
```

Comment:
Please, add spaces around **-** in this and following line: `(i - 1`
Otherwise it is good.